ecommerce_extensions/payment/processors/ednx_pp.py

```diff
@@ -0,0 +1,238 @@
+""" ednx payment processing. """
+import logging
+from decimal import Decimal
+from hashlib import sha256
+
+from oscar.apps.payment.exceptions import GatewayError, TransactionDeclined
+from oscar.core.loading import get_model
+
+try:
+    from ecommerce.extensions.payment.exceptions import InvalidSignatureError
+    from ecommerce.extensions.payment.processors import BasePaymentProcessor, HandledProcessorResponse
+except ImportError:
+    BasePaymentProcessor = object
+
+logger = logging.getLogger(__name__)
+
+PaymentEvent = get_model('order', 'PaymentEvent')
+PaymentEventType = get_model('order', 'PaymentEventType')
+ProductClass = get_model('catalogue', 'ProductClass')
+Source = get_model('payment', 'Source')
+SourceType = get_model('payment', 'SourceType')
+
+
+class ednx_pp(BasePaymentProcessor):
```

Comment:
This class name needs to be in camel case. Also, I think it might be better to use EdnxPaymentProcessor.